cmake/templates/pkgConfig.cmake.in

```diff
@@ -123,6 +123,22 @@ foreach(library ${libraries})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   elseif(${library} MATCHES "^-l")
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+  elseif(${library} MATCHES "^-")
+    # This is a linker flag/option (like -pthread)
+    # There's no standard variable for these, so create an interface library to hold it
+    if(NOT @PROJECT_NAME@_NUM_DUMMY_TARGETS)
+        set(@PROJECT_NAME@_NUM_DUMMY_TARGETS 0)
+    endif()
+    MATH(EXPR VAR "${@PROJECT_NAME@_NUM_DUMMY_TARGETS}+1")
+    add_library("catkin::@PROJECT_NAME@::dummy${@PROJECT_NAME@_NUM_DUMMY_TARGETS}" INTERFACE IMPORTED)
```

Comment:
Since the target name is visible in downstream packages I would rather prefer a different name, maybe "wrapped-linker-optionN"?

Reply:
Just `wrapped-linker-optionN` or `catkin::my-project::wrapped-linker-optionN`?

Reply:
I don't mind. While the first is unlikely to collide the second one should *really* not :wink:

Reply:
Changed target name in 8d6694a558fc94b607dee3ca9b1e4b167ffafa48